lib/cancan/ability.rb

```diff
@@ -61,15 +61,12 @@ module Ability
     #
     # Also see the RSpec Matchers to aid in testing.
     def can?(action, subject, *extra_args)
-      subject = extract_subjects(subject)
-
-      match = subject.map do |subject|
-        relevant_rules_for_match(action, subject).detect do |rule|
-          rule.matches_conditions?(action, subject, extra_args)
+      extract_subjects(subject).each do |a_subject|
```

Comment:
I think it will now abort at the first matching subject, whereas previously it would find one matching rule for each subject.